nvidia_clara/pipeline_types.py

```diff
@@ -162,7 +162,7 @@ def metadata(self, metadata: Mapping[str, str]):
 
 class PipelineInfo:
 
-    def __init__(self, pipeline_id: PipelineId = None, name: str = None, metadata: Mapping[str, str] = None):
+    def __init__(self, pipeline_id: PipelineId = None, name: str = None, metadata: Mapping[str, str] = dict()):

```

Comment:
you understand that `dict()` in place of `None` requires an allocation, that you won't likely use, to be made - right?

Reply:
The motivation is to allow users to be able to easily append to the metadata variable of an instance even after not specifying it in the constructor